txscript/consensus_test.go

```diff
@@ -244,59 +245,98 @@ func TestCheckSignatureEncoding(t *testing.T) {
 	}
 }
 
-// TestCheckPubKeyEncoding ensures the internal checkPubKeyEncoding function
-// works as expected.
+// TestCheckPubKeyEncoding ensures the CheckPubKeyEncoding function works as
+// expected.
 func TestCheckPubKeyEncoding(t *testing.T) {
 	t.Parallel()
 
 	tests := []struct {
-		name    string
-		key     []byte
-		isValid bool
-	}{
-		{
-			name: "uncompressed ok",
-			key: hexToBytes("0411db93e1dcdb8a016b49840f8c53bc1eb68" +
-				"a382e97b1482ecad7b148a6909a5cb2e0eaddfb84ccf" +
-				"9744464f82e160bfa9b8b64f9d4c03f999b8643f656b" +
-				"412a3"),
-			isValid: true,
-		},
-		{
-			name: "compressed ok",
-			key: hexToBytes("02ce0b14fb842b1ba549fdd675c98075f12e9" +
-				"c510f8ef52bd021a9a1f4809d3b4d"),
-			isValid: true,
-		},
-		{
-			name: "compressed ok",
-			key: hexToBytes("032689c7c2dab13309fb143e0e8fe39634252" +
-				"1887e976690b6b47f5b2a4b7d448e"),
-			isValid: true,
-		},
-		{
-			name: "hybrid",
-			key: hexToBytes("0679be667ef9dcbbac55a06295ce870b07029" +
-				"bfcdb2dce28d959f2815b16f81798483ada7726a3c46" +
-				"55da4fbfc0e1108a8fd17b448a68554199c47d08ffb1" +
-				"0d4b8"),
-			isValid: false,
-		},
-		{
-			name:    "empty",
-			key:     nil,
-			isValid: false,
-		},
-	}
+		name string
+		key  []byte
+		err  error
+	}{{
+		name: "uncompressed ok",
+		key: hexToBytes("04" +
+			"11db93e1dcdb8a016b49840f8c53bc1eb68a382e97b1482ecad7b148a6909a5c" +
+			"b2e0eaddfb84ccf9744464f82e160bfa9b8b64f9d4c03f999b8643f656b412a3"),
+		err: nil,
+	}, {
+		name: "compressed ok (ybit = 0)",
+		key: hexToBytes("02" +
+			"ce0b14fb842b1ba549fdd675c98075f12e9c510f8ef52bd021a9a1f4809d3b4d"),
+		err: nil,
+	}, {
+		name: "compressed ok (ybit = 1)",
+		key: hexToBytes("03" +
+			"2689c7c2dab13309fb143e0e8fe396342521887e976690b6b47f5b2a4b7d448e"),
+		err: nil,
+	}, {
+		// Although public keys not on the curve will ultimately be invalid, it
+		// is considered a valid encoding from from the standpoint of preventing
```

Comment:
Typo here `from from` (and in the following comment as well)

Reply:
Nice catch.  Fixed.